src/Http/Http/src/Features/RequestServicesFeature.cs

```diff
@@ -19,7 +16,6 @@ public class RequestServicesFeature : IServiceProvidersFeature, IDisposable
 
         public RequestServicesFeature(HttpContext context, IServiceScopeFactory scopeFactory)
         {
-            Debug.Assert(scopeFactory != null);
```

Comment:
Can't we revert this now that we throw an ArgumentNullException from the HttpContextFactory ctor?

Reply:
Sure.

Reply:
Since we're making it public, this should just throw an ArgumentNullException too.

Reply:
Well, no, we can't there's one more scenario that needs to be supported. The test scenario. The old feature was just a poco and this preserves that behavior. So if I new up a DefaultHttpContext and get or set RequestServices it preserves the old behavior. We still need to allow null.